### ISSUE TITLE
Bug: `Boxes.to_mask` on a list-backed object fills pixel (0, 0) for every padding row

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4252]
`Boxes.to_mask` exports with `to_tensor("xyxy", as_padded_sequence=True)`, and the zero padding row of a list-backed object exports as the exclusive `xyxy` box `[0, 0, 1, 1]` under the inclusive `+1` convention (the "unit box" padding behavior recorded on the `Boxes` anchor since #4060). The mask then marks pixel `(0, 0)` in the channel of every padding entry, so the returned `(B, N_max, H, W)` mask carries a spurious pixel per padded slot instead of an empty channel. This is the first concrete consumer of the padded-row value; it is the same family as #4179 (padding rows exported as boxes after indexing).

Reproduced 2026-09-05 at `1eea5835`, PyTorch 2.14.0, CPU and MPS.

## Reproduction

```python
import torch
from kornia.geometry.boxes import Boxes

box = torch.tensor([[[1., 1.], [2., 1.], [2., 2.], [1., 2.]]])
boxes = Boxes([box, torch.cat([box, box + 2.0])])      # first sample has one box, second has two
mask = boxes.to_mask(5, 5)
print(mask.shape, "padding channel:", mask[0, 1].nonzero().tolist())
print(boxes.to_tensor("xyxy", as_padded_sequence=True)[0, 1].tolist())
```

## Observed

```text
torch.Size([2, 2, 5, 5]) padding channel: [[0, 0]]
[0.0, 0.0, 1.0, 1.0]
```

## Expected

An empty mask channel for a padding entry (or no channel at all, if `to_mask` should honor the list cardinality the way `to_tensor` does without `as_padded_sequence`). Either choice depends on the padded-row value contract, which is not settled: the batch-4a audit recorded it as a design candidate because the padded row's value after mode conversion is unspecified. Batch 4c documents the pixel on `to_mask` and pins it (`test_wart_to_mask_fills_the_origin_pixel_for_list_padding_rows_4252`).

## Classification

Broken output for list-backed input (a padding artifact surfaces as data). Idiom 3, inherited from the padded-row conversion rather than from `to_mask` itself.

## Disposition

n/a — single symbol

The fix is coupled to the padded-row contract and to #4179; whoever settles that contract should close both.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
